presto-main/src/main/java/com/facebook/presto/sql/planner/LogicalPlanner.java

```diff
@@ -130,29 +130,35 @@ public Plan plan(Analysis analysis, Stage stage)
 
     public PlanNode planStatement(Analysis analysis, Statement statement)
     {
-        if (statement instanceof CreateTableAsSelect) {
+        if (statement instanceof CreateTableAsSelect && analysis.isCreateTableAsSelectNoOp()) {
             checkState(analysis.getCreateTableDestination().isPresent(), "Table destination is missing");
+            List<Expression> emptyRow = ImmutableList.of();
+            PlanNode source = new ValuesNode(idAllocator.getNextId(), ImmutableList.of(), ImmutableList.of(emptyRow));
+            return new OutputNode(idAllocator.getNextId(), source, ImmutableList.of(), ImmutableList.of());
```

Comment:
How about moving this line into `createOutputPlan` inside a `if (analysis.isCreateTableAsSelectNoOp())` check?